app/models/serial_search.rb

```diff
@@ -25,6 +25,14 @@ def self.min_request_time
     8.hours.ago
   end
 
+  def self.text_search(query)
+    if query.present?
+      where('serial @@ :q', q: query)
```

Comment:
If you're only including one parameter, you don't need to name it (at least that's the convention that I follow).

So I'd do `where('serial @@ ?', query)`